sklearn/utils/_array_api.py

```diff
@@ -60,6 +60,8 @@ def yield_namespace_device_dtype_combinations(include_numpy_namespaces=True):
     """Yield supported namespace, device, dtype tuples for testing.
 
     Use this to test that an estimator works with all combinations.
+    Use in conjunction with `ids=_get_namespace_device_dtype_ids` to give
```

Comment:
Can `yield_namespace_device_dtype_combinations` out the ids so we do not need to use `_get_namespace_device_dtype_ids` anymore?

Concretely, in `yield_namespace_device_dtype_combinations`:

```
yield pytest.param(array_namespace, array_api_strict.Device("CPU_DEVICE"), "float64", id="CPU_DEVICE")
```

Reply:
That would be a much better option! Didn't think of it. I'll try it out.

(also is it possible to 'undo' a commit? i.e. https://github.com/scikit-learn/scikit-learn/pull/31074)

Reply:
Assuming you are branching off main, the following will create a commit that will undo a commit:

```bash
git revert 677869070fb1f2f7a6489fa2eedde03acf58c4e2
```

`677869070fb1f2f7a6489fa2eedde03acf58c4e2` is the commit for #31074